### DISCUSSION TITLE
Is it possible to get Windhawk to work before windows boots?

### DISCUSSION THREAD

**Discussion by happymimimix** [ramensoftware/windhawk#346] [Ideas]
I wonder if its possible to get windhawk to work before windows boots? Maybe a native edition of windhawk that can be used to patch things during boot time. 
I prefer something that can works correctly with legacy boot. Please don't give me an .efi file that patch files. Personally, I love legacy boot well over UEFI. I would always ensure that the device I purchase has legacy boot emulation function in their EFI firmware, and I always install Windows with legacy boot, even Windows 11. Even when my device supports TPM, I would still use the registry hack to disable TPM checks in windows 11's installer and install it in legacy.

**Comment by m417z** [comment]
Currently, Windhawk uses a service. [Here it was suggested](https://windhawk.net/mods/classic-theme-enable-with-extended-compatibility#:~:text=6.%20How%20to%20configure%20system%20start%20of%20Windhawk) to make another service depend on Windhawk to make Windhawk start earlier, and [here's a discussion about which service to use for that](https://github.com/ramensoftware/windhawk-mods/pull/917#:~:text=in%20the%20future.-,FYI%2C%20I%20saw%20somebody%20on%20Discord%20recommending%20to%20make%20ProfSvc%20depend%20on%20Windhawk.%20I%20didn%27t%20try%20it%2C%20but%20it%20worked%20for%20them.,-I%20imagine%20that). But that's the best Windhawk has to offer right now, it doesn't have the ability to start before Windows boots, which will likely require a kernel driver or some other low-level mechanism.

**Reply by happymimimix** [reply]
I mean, make a native version of windhawk. Something similar to memtest.exe.

**Reply by m417z** [reply]
That's not something I plan on working on in the near future, but I'll keep the suggestion in mind. Maybe on day...

**Reply by happymimimix** [reply]
> That's not something I plan on working on in the near future, but I'll keep the suggestion in mind. Maybe on day...

I want to tweak the windows boot screen. It would be fun if windhawk could do something like this.

**Comment by silverbacknet** [comment]
I'm completely unclear on what your goal here is, not the what, but *why* you want it to run on bare metal. Windhawk modifies the behavior of *Windows*, so there's nothing for it to do if there's no Windows loaded.

**Reply by silverbacknet** [reply]
If you mean that you want it loaded as early in the boot process as possible, then you need a kernel driver, not a bare metal executable.

**Reply by happymimimix** [reply]
> I'm completely unclear on what your goal here is, not the what, but _why_ you want it to run on bare metal. Windhawk modifies the behavior of _Windows_, so there's nothing for it to do if there's no Windows loaded.

Modify the windows boot screen. Without touch files on disk.
